libkbfs/key_manager.go

```diff
@@ -450,21 +450,41 @@ func (km *KeyManagerStandard) Rekey(ctx context.Context, md *RootMetadata, promp
 			return false, nil, err
 		}
 		resolvedInfo := resolvedHandle.ConflictInfo()
-		if latestHandle.ConflictInfo != resolvedInfo {
-			km.log.CDebugf(ctx, "handle for %s is conflicted",
-				handle.GetCanonicalPath())
-			if resolvedInfo != nil {
-				// ConflictInfo is inconsistent.
+		if resolvedInfo == nil {
+			if latestHandle.ConflictInfo == nil {
+				// Nothing to do.
+			} else {
+				km.log.CDebugf(
+					ctx, "handle for %s is conflicted",
+					handle.GetCanonicalPath())
+				// Set the conflict info in the resolved handle.
+				// This will get stored in the metadata block.
+				// TODO: We should do some verification of this.
+				resolvedHandle.SetConflictInfo(latestHandle.ConflictInfo)
+			}
+		} else {
+			if latestHandle.ConflictInfo == nil {
+				// Conflict info disappeared?
+				err := TlfHandleExtensionMismatchError{
+					Expected: resolvedInfo,
+					Actual:   nil,
+				}
+				return false, nil, err
+			}
+			// Make sure conflict info is the same.
+			equal, err := CodecEqual(
+				km.config.Codec(), resolvedInfo,
+				latestHandle.ConflictInfo)
```

Comment:
Pointer comparisons (the old method) won't work, so we need a deep comparison. But since `ConflictInfo` may contain unknown fields, we need to use `CodecEqual` for the deep comparison.